kotlinpoet/src/commonMain/kotlin/com/squareup/kotlinpoet/NameAllocator.kt

```diff
@@ -77,7 +77,39 @@ public class NameAllocator private constructor(
   private val allocatedNames: MutableSet<String>,
   private val tagToName: MutableMap<Any, String>,
 ) {
-  public constructor() : this(mutableSetOf(), mutableMapOf())
+  public constructor() : this(preAllocateKeywords = true)
+
+  /**
+   * @param preAllocateKeywords If true, all Kotlin keywords will be pre-allocated and suffixed with
+   * underscores to avoid using them as identifiers:
```

Comment:
Split this into two sentences. Currently it sounds like the preallocated names receive the underscore, but I'm fact it's the requested names which collide with keywords that receive it.

Reply:
Done. [b9d801e](https://github.com/square/kotlinpoet/pull/1803/commits/b9d801e32fffdcd8483310091e2def49d9edb6b1)